controllers/install/armadaserver_controller.go

```diff
@@ -0,0 +1,359 @@
+
+	deletionTimestamp := as.ObjectMeta.DeletionTimestamp
+	// examine DeletionTimestamp to determine if object is under deletion
+	if deletionTimestamp.IsZero() {
+		// The object is not being deleted, so if it does not have our finalizer,
+		// then lets add the finalizer and update the object. This is equivalent
+		// registering our finalizer.
+		if !controllerutil.ContainsFinalizer(&as, asFinalizer) {
+			logger.Info("Attaching finalizer to ArmadaServer object", "finalizer", asFinalizer)
+			controllerutil.AddFinalizer(&as, asFinalizer)
+			if err := r.Update(ctx, &as); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		logger.Info("ArmadaServer object is being deleted", "finalizer", asFinalizer)
+		// The object is being deleted
+		if controllerutil.ContainsFinalizer(&as, asFinalizer) {
+			// our finalizer is present, so lets handle any external dependency
+			logger.Info("Running cleanup function for ArmadaServer object", "finalizer", asFinalizer)
+
+			// remove our finalizer from the list and update it.
+			logger.Info("Removing finalizer from ArmadaServer object", "finalizer", asFinalizer)
+			controllerutil.RemoveFinalizer(&as, asFinalizer)
+			if err := r.Update(ctx, &as); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+
+		// Stop reconciliation as the item is being deleted
+		return ctrl.Result{}, nil
+	}
+
+	mutateFn := func() error { return nil }
+
+	logger.Info("Upserting ArmadaServer Deployment object")
+	if components.Deployment != nil {
+		if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, components.Deployment, mutateFn); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	logger.Info("Upserting ArmadaServer Ingress object")
+	if components.Ingress != nil {
+		if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, components.Ingress, mutateFn); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	logger.Info("Upserting ArmadaServer Ingress_Rest object")
+	if components.Ingress_Rest != nil {
+		if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, components.Ingress_Rest, mutateFn); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	logger.Info("Upserting ArmadaServer Service object")
+	if components.Service != nil {
+		if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, components.Service, mutateFn); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	logger.Info("Upserting ArmadaServer ServiceAccount object")
+	if components.ServiceAccount != nil {
+		if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, components.ServiceAccount, mutateFn); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	logger.Info("Upserting ArmadaServer Secret object")
+	if components.Secret != nil {
+		if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, components.Secret, mutateFn); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	logger.Info("Upserting ArmadaServer PodDisruptionBudget object")
+	if components.Secret != nil {
```

Comment:
```suggestion
	if components.PodDisruptionBudget != nil {
```

Reply:
This and following component conditional tests fixed in commit 29a2d25.